### PR TITLE
feat(echo): add Echo kustomizations

### PR DESCRIPTION
**Pull request by maggieneterval**
- feat(echo): add Echo kustomizations

  I stood on the shoulders of gurus for this change; going to add a few questions inline in places where I wasn't 100% sure I should be blindly copy/pasting from the Clouddriver config. 🙈

### CODE REVIEW

File: base/echo/base/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: echo
+        app.kubernetes.io/instance: echo
+    spec:
+      containers:
+      - env:
+        - name: JAVA_OPTS
+          value: -XX:MaxRAMPercentage=80.0
```
Comment (maggieneterval) [inline base/echo/base/deployment.yml:23]: Do we think this increase from the Halyard-set default is okay for Echo as well as Clouddriver?

File: base/echo/base/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          protocol: TCP
+        readinessProbe:
+          httpGet:
+            port: traffic-port
+            path: /health
+          failureThreshold: 3
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+        resources:
```
Comment (maggieneterval) [inline base/echo/base/deployment.yml:40]: I also blindly copied this `resources` block from Clouddriver; even though it's empty on Halyard-deployed Echo. Were these values meant to be Clouddriver-specific or reasonable for all services?

File: base/echo/base/deployment.yml (hunk 1 above)
Comment (ezimanyi) [reply base/echo/base/deployment.yml:23]: Yes, I deliberately increased it as it seemed silly to limit to 50% of the RAM when running in a container with nothing else there, so I'd suggest doing this for all the services.

File: base/echo/base/deployment.yml (hunk 2 above)
Comment (ezimanyi) [reply base/echo/base/deployment.yml:40]: These will probably need some tuning---they're probably the most common things people will configure themselves, but we should still try to have reasonable defaults.  My guess is maybe we could go a bit lower for `echo`?  But maybe what we should do after everything deploys is make a reasonable size deployment and see how good our estimates are, and tune it then...at least as best as we can for defaults.

File: base/echo/echo-scheduler/profile.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: apps/v1
```
Comment (ezimanyi) [inline base/echo/echo-scheduler/profile.yml:1]: It looks like these two `profile.yml` files aren't being referenced from the `kustomization`.yml` files, so probably aren't doing anything.

Comment (maggieneterval) [reply base/echo/echo-scheduler/profile.yml:1]: 🤦Great catch, I guess I deployed two over-achieving echos BOTH doing scheduling...

File: base/echo/base/deployment.yml (hunk 2 above)
Comment (maggieneterval) [reply base/echo/base/deployment.yml:40]: Okay that sounds great; I'll leave these for now but I added a todo in our kleat doc to edit these based on tuning when we've completed all the services.

**Review by ezimanyi — APPROVED** [review]
🔊〰️ 🔊〰️ 🔊〰️ 🔊〰️👂
💯 🥳 🥇 🤗

**Pull request merged**